### PR TITLE
[Backport release-1.25] Switch from k8s.gcr.io to registry.k8s.io

### PR DESCRIPTION
**Pull request by twz123**
## Description

Backport of #2346.

Kubernetes switched to the new registry in k8s 1.25. Do the same for k0s.

Reference: kubernetes/k8s.io#1834 and kubernetes/kubernetes#109938.

This also fixes a discrepancy between the default pause image used in containerd and the one used by k0s.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

- [ ] Manual test
- [ ] Auto test added

## Checklist:

- [x] My code follows the style [guidelines](https://github.com/k0sproject/k0s/blob/main/docs/contributors/overview.md) of this project 
- [x] My commit messages are [signed-off](https://github.com/k0sproject/k0s/blob/main/docs/contributors/github_workflow.md)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: inttest/airgap/airgap_test.go [hunk 1]
```diff
@@ -36,7 +36,7 @@ spec:
 const etcHosts = `
 127.0.0.8 docker.io
 127.0.0.8 gcr.io
-127.0.0.8 k8s.gcr.io
+127.0.0.8 registry.k8s.io
```
Comment (jnummelin) [inline inttest/airgap/airgap_test.go:39]: I'd also leave `k8s.gcr.io` into the "block" list to be sure we block any accidental access to it too

File: docs/configuration.md [hunk 2]
```diff
@@ -84,17 +84,17 @@ spec:
...
+      version: v0.6.1
     kubeproxy:
-      image: k8s.gcr.io/kube-proxy
+      image: registry.k8s.io/kube-proxy
       version: v1.25.4
     coredns:
-      image: k8s.gcr.io/coredns/coredns
-      version: v1.7.0
+      image: docker.io/coredns/coredns
+      version: 1.9.4
```
Comment (jnummelin) [inline docs/configuration.md:97]: Do we want to bump up coreDNS this much in a backport?

**Review by jnummelin — COMMENTED** [review]
I don't think we can backport this as-is to 1.24 as that uses very different image versions

File: inttest/airgap/airgap_test.go (hunk 1 above)
Comment (twz123) [reply inttest/airgap/airgap_test.go:39]: I'd like to file a separate PR for this, as this is a backport, which wouldn't be on par with the original PR. Alternatively, I can add a commit here, which we then cherry-pick back to main.

File: docs/configuration.md (hunk 2 above)
Comment (twz123) [reply docs/configuration.md:97]: This is the docs, which have been updated in the original PR to match the versions that are actually used by k0s. The 1.25 release is currently on 1.9.4.

Comment (jnummelin) [reply docs/configuration.md:97]: ah 🤦  all good then :)

File: inttest/airgap/airgap_test.go (hunk 1 above)
Comment (jnummelin) [reply inttest/airgap/airgap_test.go:39]: good point. separate PR makes sense.

**Pull request merged**